scene/2d/tile_map.cpp

```diff
@@ -362,11 +362,11 @@ void TileMapLayer::_rendering_update() {
 
 					// Random animation offset.
 					real_t random_animation_offset = 0.0;
-					if (atlas_source->get_tile_animation_mode(cell_data.cell.get_atlas_coords()) != TileSetAtlasSource::TILE_ANIMATION_MODE_DEFAULT) {
+					if ((atlas_source->get_tile_animation_mode(cell_data.cell.get_atlas_coords()) == TileSetAtlasSource::TILE_ANIMATION_MODE_RANDOM_START_TIMES)|| (atlas_source->get_tile_animation_mode(cell_data.cell.get_atlas_coords()) == TileSetAtlasSource::TILE_ANIMATION_MODE_BOUNCE_RANDOM_START_TIMES)) {
 						Array to_hash;
 						to_hash.push_back(local_tile_pos);
 						to_hash.push_back(get_instance_id()); // Use instance id as a random hash
-						random_animation_offset = RandomPCG(to_hash.hash()).randf();
+						random_animation_offset = RandomPCG(to_hash.hash()).randf() * 86400.0;
```

Comment:
You've included your other change from your other PR, you need to only have the changes of this PR